connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -431,6 +432,7 @@ public interface LiteralSupplier<T> {
      * @param args the arguments
      */
     public static void main(String[] args) {
+        LogManager.shutdown();
```

Comment:
why we need this change?

Reply:
In the current design, the way we generate our documentation file is executing the `main()` in above 434 lines and took everything from the stdout as a file. 

```gradle
task genConsumerMetricsDocs(type: JavaExec) {
    classpath = sourceSets.test.runtimeClasspath
    mainClass = 'org.apache.kafka.clients.consumer.internals.ConsumerMetrics'
    if( !generatedDocsDir.exists() ) { generatedDocsDir.mkdirs() }
    standardOutput = new File(generatedDocsDir, "consumer_metrics.html").newOutputStream()
  }
```
This caused all log information also be added into our docs as you can see here

https://kafka.apache.org/documentation.html#connect_monitoring
![image](https://github.com/apache/kafka/assets/38662781/d9ec6023-6b0c-4165-b6f8-59cb039669d8)

Reply:
Since the log information is generated by the Metric class, another way I found is to add `log4j.logger.org.apache.kafka.common.metrics.Metrics=WARN` to the `/kafka/connect/runtime/src/test/resources/log4j.properties`. 

However, this way also affects the log behavior during the test execution so I think only close the log in main function is a better way.

Reply:
Do other doc tasks have similar issue? for example: `genSourceConnectorConfigDocs`, `genSinkConnectorConfigDocs`, etc.

Reply:
Nice catch @KevinZTW !

Reply:
After executing those tasks, I think they don't have similar issue and the logger implementation couldn't be loaded
```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
```
And I think it's because their classpath set to the `sourceSets.main.runtimeClasspath` so instead of close the logger directly I would leverage this to eliminate unexpected log info

Reply:
> And I think it's because their classpath set to the sourceSets.main.runtimeClasspath so instead of close the logger directly I would leverage this to eliminate unexpected log info

pardon me, why this works?

Reply:
Sorry took a while to check this. I think it is because usually in our project, the `StaticLoggerBinder` is provided by `slf4j-reload4j` and In the Gradle build file, the project `:connect:runtime` adds that as test runtime only. 

Hence, if we specify the `classpath` as `main` runtime, the path where `slf4j-reload4j`'s jar file is located won't be included. Hence cause the SL4J change to use NOP logger implementation

Reply:
Thanks for sharing. That is interesting.